### PR TITLE
4302 add more test case in storage package

### PR DESCRIPTION
**Pull request by NNagato**

### CODE REVIEW

File: v3/common/errors.go [hunk 1]
```diff
@@ -36,4 +36,10 @@ var (
 	ErrQuoteAssetInvalid = errors.New("quote asset is invalid")
 	// ErrBaseAssetInvalid is returned when base asset doesn't contain exchange id in the  create/update trading  request
 	ErrBaseAssetInvalid = errors.New("base asset is invalid")
+	// ErrExchangeIDNotExists is returned when exchange id is not exist.
+	ErrExchangeIDNotExists = errors.New("exchange id is not exist")
+	// ErrAssetIDNotExists is returned when foreign key is not exist.
+	ErrAssetIDNotExists = errors.New("asset id is not exist")
+	// ErrDuplicateExchangeIDAssetID is returned when unique constraint is violated.
+	ErrDuplicateExchangeIDAssetID = errors.New("duplicate key value violates unique constraint exchange id, asset id")
```
Comment (ducquangkstn) [inline v3/common/errors.go]: use ErrAssetExchangeAlreadyExist

File: v3/storage/postgres/trading_pair_test.go [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+			amountLimitMin:  1000,
+			amountLimitMax:  10000,
+			priceLimitMin:   0.1,
+			priceLimitMax:   10.10,
+			minNotional:     0.001,
+			assertFn: func(t *testing.T, id uint64, err error) {
+				require.Zero(t, id)
+				require.NotNil(t, err)
+			},
+		},
```
Comment (ducquangkstn) [inline v3/storage/postgres/trading_pair_test.go]: add logic when add to trading pair with quote_id referring an asset with is_quote=false
Note: 
https://github.com/KyberNetwork/reserve-data/blob/v3-develop/v3/storage/postgres/sql.go#L421

File: v3/storage/postgres/assets_exchange_test.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+			minDeposit:        0.0001,
+			withdrawFee:       0.0001,
+			targetRecommended: 0.001,
+			targetRatio:       0.001,
+
+			assertFn: func(t *testing.T, id uint64, err error) {
+				assert.Zero(t, id)
+				require.Equal(t, commonv3.ErrAssetIDNotExists, err)
+			},
+		},
```
Comment (ducquangkstn) [inline v3/storage/postgres/assets_exchange_test.go]: add test for those constraint if possible:
https://github.com/KyberNetwork/reserve-data/blob/v3-develop/v3/storage/postgres/sql.go#L64

Comment (NNagato) [reply v3/storage/postgres/assets_exchange_test.go]: added more test in https://github.com/KyberNetwork/reserve-data/pull/547/commits/f7b083e3d607e42ba396ea0a553a584525443bf4

File: v3/storage/postgres/trading_pair_test.go (hunk 2 above)
Comment (NNagato) [reply v3/storage/postgres/trading_pair_test.go]: added in https://github.com/KyberNetwork/reserve-data/pull/547/commits/f7b083e3d607e42ba396ea0a553a584525443bf4

File: v3/common/errors.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (NNagato) [reply v3/common/errors.go]: fixed in https://github.com/KyberNetwork/reserve-data/pull/547/commits/98a863bd9f6b2c7a981bf38b20f04ebabf9adad8

File: v3/common/errors.go [hunk 4]
```diff
@@ -36,4 +36,10 @@ var (
 	ErrQuoteAssetInvalid = errors.New("quote asset is invalid")
 	// ErrBaseAssetInvalid is returned when base asset doesn't contain exchange id in the  create/update trading  request
 	ErrBaseAssetInvalid = errors.New("base asset is invalid")
+	// ErrExchangeIDNotExists is returned when exchange id is not exist.
+	ErrExchangeIDNotExists = errors.New("exchange id is not exist")
```
[the remark this answers is not in the document]
Comment (NNagato) [reply v3/common/errors.go]: fixed in https://github.com/KyberNetwork/reserve-data/pull/547/commits/98a863bd9f6b2c7a981bf38b20f04ebabf9adad8

File: v3/common/errors.go (hunk 1 above)
Comment (NNagato) [reply v3/common/errors.go]: fixed in https://github.com/KyberNetwork/reserve-data/pull/547/commits/98a863bd9f6b2c7a981bf38b20f04ebabf9adad8

**Comment by secmask** [comment]
@NNagato please resolve conflict

**Pull request merged**